engine/evd.ml

```diff
@@ -139,6 +139,26 @@ module Abstraction = struct
   let abstract_last l = Abstract :: l
 end
 
+module Identity :
+sig
+  type t
+  val make : econstr list -> t
+  val none : unit -> t
+  val repr : named_context_val -> Filter.t -> t -> econstr list
+  val is_identity : econstr list -> t -> bool
+end =
+struct
+  type t = econstr list option ref
```

Comment:
Is the ref mutated somewhere? I can't see it.

Reply:
Wut, you're right. Forgot to mutate it at all :crayon: I'll have to rerun a bench then, the perf gain is probably even better than that...

Reply:
That said in the latest version of the code, the instance is mostly shared with the context. Since it's unlikely to have an non-empty filter it might not make a lot of difference in the end.